overlord/servicestate/servicestate.go

```diff
@@ -119,3 +122,80 @@ func Control(st *state.State, appInfos []*snap.AppInfo, inst *Instruction, conte
 
 	return tts, nil
 }
+
+// StatusDecorator supports decorating client.AppInfos with service status.
+type StatusDecorator struct {
+	sysd systemd.Systemd
+}
+
+// NewStatusDecorator returns a new StatusDecorator.
+func NewStatusDecorator(rep interface {
+	Notify(string)
+}) *StatusDecorator {
+	return &StatusDecorator{
+		sysd: systemd.New(dirs.GlobalRootDir, systemd.SystemMode, rep),
+	}
+}
+
+// DecorateWithStatus adds service status information to the given client.AppInfo associated with the given snap.AppInfo.
+func (sd *StatusDecorator) DecorateWithStatus(appInfo *client.AppInfo, snapApp *snap.AppInfo) error {
+	if appInfo.Snap != snapApp.Snap.InstanceName() || appInfo.Name != snapApp.Name {
+		return fmt.Errorf("internal error: misassociated app info %v and client app info %s.%s", snapApp, appInfo.Snap, appInfo.Name)
+	}
+	if !snapApp.IsService() {
+		// nothing to do
+		return nil
+	}
```

Comment:
Should we check: `if !snapApp.IsActive() { return nil }` here too?

Reply:
yes, it's a bit safer

Reply:
this needed test changes to have IsActive return true.